app/controllers/change_requests_controller.rb

```diff
@@ -53,6 +71,74 @@ def create_change_request
     render status: :created, json: ChangeRequestsPresenter.present(@change_request)
   end
 
+  def handle_insert
+
+    change_request = params[:change_request]
+    case change_request[:type]
+    when "addresses"
+      address = Address.new
+      change_request_params = params.require(:change_request).permit(field_changes: [
+          :attention,
+          :city,
+          :address_1,
+          :address_2,
+          :address_3,
+          :address_4,
+          :city,
+          :state_province,
+          :postal_code,
+          :country,
+          :online,
+          :region,
+          :name,
+          :description,
+          :transportation,
+      ])
+      change_request_params.require(:field_changes)
+      address.attributes = change_request_params[:field_changes]
+      address.resource_id = change_request[:resource_id]
+
+      begin
+        a = Geokit::Geocoders::GoogleGeocoder.geocode address.address_1 + "," + address.city + "," + address.state_province
+        address.latitude = a.latitude
+        address.longitude = a.longitude
+      rescue => error
+        puts "google geocoding failed for address " + address.id.to_s + ": " + error.message
+      end
+
+      address.save!
+
+      @change_request = AddressChangeRequest.create(object_id: address.id, resource_id: Address.find(address.id).resource_id)
+      @change_request.field_changes = field_changes
+      persist_change (@change_request)
```

Comment:
I just realized that this line is here, which is a bit confusing because we're saving the Address twice: once just a few lines higher up, and once here, after we construct the `@change_request` object.

@jjfreund, is there a reason you created the Address object above with all of the field changes saved before you call `persist_change`, which modifies the Address object again with the exact same field changes before saving it again?

It also looks like you're not saving the ChangeRequest's field changes to the database, which means that after it calls `persist_change`, it throws out the field change data from the ChangeRequest object after updating the object itself.

For consistence with the update methods and to eventually support the case where we gate edits by a moderator, I'd actually prefer to only create and save the address info after creating the ChangeRequest, using only info from the ChangeRequest. That would allow us to decouple in time the creation of the ChangeRequest and the approval of the ChangeRequest.